android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java

```diff
@@ -308,26 +321,62 @@ private void setAvatarDataForContactIfAvailable(Contact contact) {
     }
   }
 
-  private void loadContactPhotoHighRes(Contact contact, boolean photoHighResolution) {
-      try {
-          Uri uri = ContentUris.withAppendedId(ContactsContract.Contacts.CONTENT_URI, Long.parseLong(contact.identifier));
-          InputStream input = ContactsContract.Contacts.openContactPhotoInputStream(contentResolver, uri, photoHighResolution);
-
-          if (input == null){
-              contact.avatar = new byte[0];
-              return;
-          }
+  private void getAvatars(final List<String> identifiers, final boolean highRes,
+      final Result result) {
+    new GetAvatarsTask(identifiers, highRes, contentResolver, result).execute();
```

Comment:
I would suggest using `executeOnExecutor()` variant of this so you can control how many threads are spawned.

Reply:
Changed to use a `ThreadPoolExecutor`. I tried using `Executors.newCachedThreadPool()` but a stress test with 1000 contacts crashed the app. Tweaking it a bit, I ended up with:

```java
new ThreadPoolExecutor(
    0 /* corePoolSize */,
    10 /* maximumPoolSize */,
    60 /* keepAliveTime */,
    TimeUnit.SECONDS /* keepAliveTimeUnit */,
    new ArrayBlockingQueue<Runnable>(1000) /* workQueue */
)
```

This was able to process 1000 contacts without crashing, albeit somewhat slowly. I figure that:

1. Humans are unlikely to have that many contacts on their device.
1. If a client application is dealing with more contacts than that, then they should be paginating their `getAvatar()` requests to avoid this problem.

Let me know if there is a more appropriate executor to use here, as I'm not all that familiar with them myself. I also updated the existing `GetContacts` task to use the same executor, as I figure that will hopefully keep the cache warm until avatars are lazily fetched.